Makefile

```diff
@@ -149,56 +88,32 @@ icu-macos-fix-install:
 	brew install icu4c
 .PHONY: icu-macos-fix-install
 
-icu-macos-fix-build:
-	stack --stack-yaml=stack-8.2.1.yaml build text-icu     \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include    \
-	&& stack --stack-yaml=stack-8.2.2.yaml build text-icu  \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include    \
-	&& stack --stack-yaml=stack-8.4.2.yaml build text-icu  \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include    \
-	&& stack --stack-yaml=stack-8.4.3.yaml build text-icu  \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include    \
-	&& stack --stack-yaml=stack-8.4.4.yaml build text-icu  \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include    \
-	&& stack --stack-yaml=stack-8.6.1.yaml build text-icu  \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include    \
-	&& stack --stack-yaml=stack-8.6.2.yaml build text-icu  \
-	  --extra-lib-dirs=/usr/local/opt/icu4c/lib            \
-	  --extra-include-dirs=/usr/local/opt/icu4c/include
+icu-macos-fix-build: $(foreach version, $(GHC_VERSIONS), icu-macos-fix-build-$(version))
 .PHONY: icu-macos-fix-build
 
+icu-macos-fix-build-%:
+	stack --stack-yaml=stack-$*.yaml build text-icu \
+	  --extra-lib-dirs=/usr/local/opt/icu4c/lib \
+	  --extra-include-dirs=/usr/local/opt/icu4c/include
+
 # ------------------------------------------------------
 
 HIE_GIT_REF:=$(shell git describe --tags)
-HIE_DIST_NAME:=hie-${HIE_GIT_REF}-`uname -m`-`uname -s`
+HIE_DIST_NAME:=hie-${HIE_GIT_REF}-$(shell uname -m)-$(shell uname -s)
 HIE_DIST_DIR:=/tmp/${HIE_DIST_NAME}
+
 ## Creates a tarball containing all the hie binaries
-dist:
-	mkdir ${HIE_DIST_DIR}
-	stack --stack-yaml=stack-8.2.1.yaml build
-	cp .stack-work/install/*/*/8.2.1/bin/hie ${HIE_DIST_DIR}/hie-8.2.1
-	stack --stack-yaml=stack-8.2.2.yaml build
-	cp .stack-work/install/*/*/8.2.2/bin/hie ${HIE_DIST_DIR}/hie-8.2.2
-	stack --stack-yaml=stack-8.4.2.yaml build
-	cp .stack-work/install/*/*/8.4.2/bin/hie ${HIE_DIST_DIR}/hie-8.4.2
-	stack --stack-yaml=stack-8.4.3.yaml build
-	cp .stack-work/install/*/*/8.4.3/bin/hie ${HIE_DIST_DIR}/hie-8.4.3
-	stack --stack-yaml=stack-8.4.4.yaml build
-	cp .stack-work/install/*/*/8.4.4/bin/hie ${HIE_DIST_DIR}/hie-8.4.4
-	stack --stack-yaml=stack-8.6.1.yaml build
-	cp .stack-work/install/*/*/8.6.1/bin/hie ${HIE_DIST_DIR}/hie-8.6.1
-	stack --stack-yaml=stack-8.6.2.yaml build
-	cp .stack-work/install/*/*/8.6.2/bin/hie ${HIE_DIST_DIR}/hie-8.6.2
-	cp .stack-work/install/*/*/8.6.2/bin/hie ${HIE_DIST_DIR}/hie
-	cp .stack-work/install/*/*/8.6.2/bin/hie-wrapper ${HIE_DIST_DIR}/hie-wrapper
+dist: $(foreach version, $(GHC_VERSIONS), icu-macos-fix-build-$(version))
```

Comment:
Should this have a dependency on `dist-$(version)`?

Reply:
oops, thanks for noticing

Reply:
done